google_guest_agent/addresses_test.go

```diff
@@ -28,18 +28,19 @@ func TestCompareRoutes(t *testing.T) {
 	}{
 		// These should return toAdd:
 		// In Md, not present
-		{nil, []string{"1.2.3.4/32"}, []string{"1.2.3.4/32"}, nil},
-		{nil, []string{"1.2.3.4/32", "5.6.7.8/32"}, []string{"1.2.3.4/32", "5.6.7.8/32"}, nil},
+		{nil, []string{"1.2.3.4"}, []string{"1.2.3.4"}, nil},
```

Comment:
To properly test this change, shouldn't we have test cases in the TestAddressDiff function where the metadata server returns IPs with subnetworks specified? It seems like we're missing any testing for subnetworks in this file (`/32` and others).

Reply:
The metadata server may return addresses with or without subnet masks depending on what is specified by the user, and for host local routes in some cases uses `/32` notation and in other cases does not. that's the revelation prompting this PR.

The only case where we need to modify anything provided by the metadata server is the `/32` case, since `1.1.1.1` and `1.1.1.1/32` are synonymous but won't be compared as synonymous since they aren't the same string. to handle this one case, we strip `/32` from all addresses prior to doing comparisons. 

`TestAddressDiff()` asks whether anything has changed from the previous instance of metadata to the current instance of metadata, checking the enablement logic, and doesn't seem relevant to this PR.

Reply:
> The metadata server may return addresses with or without subnet masks depending on what is specified by the user, and for host local routes in some cases uses /32 notation and in other cases does not. that's the revelation prompting this PR.

We should have tests where we parse the content of the metadata server with and without the subnet mask specified.

Reply:
the metadata parsing isn't changing with this PR. please elaborate what test you think we need here